acpi/src/lib.rs

```diff
@@ -95,18 +117,19 @@ pub enum AcpiError {
     InvalidDsdtAddress,
     InvalidMadt(MadtError),
     InvalidGenericAddress,
+
+    AllocError,
 }
 
+/// Type capable of enumerating the existing ACPI tables on the system.
+///
+/// ### Remark: If using the `allocator_api` is an option, `PlatformInfo::new()` should be used
```

Comment:
I'm not sure separating out these remarks makes stuff clearer. The wording could also be improved, just from the point that you still need to make an `AcpiTable` to get a `PlatformInfo`, but I definitely agree with pointing users towards `PlatformInfo` if they can use it.

Reply:
Perhaps this?
```
/// Type capable of enumerating the existing ACPI tables on the system.
///
/// 
/// ### Implementation Note
///
/// When using the `allocator_api` feature, [`PlatformInfo::new()`] provides a much cleaner
/// API for enumerating ACPI structures once an `AcpiTables` has been constructed.
```

Reply:
Yep, that would be fine!